### PR TITLE
Shuffle metrics 4/4: Remove bespoke diagnostics

### LINKED ISSUE #7943: Use metering for P2P shuffling instrumentation
The P2P extensions (primarily the buffers) are implementing their own version for diagnostics, e.g. here https://github.com/dask/distributed/blob/429ef8cc682da36017a28e061d1c773066470fe3/distributed/shuffle/_buffer.py#L267-L272

I think P2P diagnostics would greatly benefit from the metering / fine grained metrics and we should consider replacing (os extending) the custom ctx managers there with the metrics.meter (mind that there is also a special dashboard for P2P shuffling that is using this custom diagnostics. I don't want to break this dashboard).

cc @crusaderky @hendrikmakait

### PR DESCRIPTION
**Pull request by crusaderky**
- Closes #7943 

Please read: https://github.com/dask/distributed/issues/7943#issuecomment-1818008425
There are four commits in this PR. All but the last are the previous PRs in the chain.

### CODE REVIEW

File: distributed/dashboard/components/scheduler.py [hunk 1]
```diff
@@ -4326,16 +4326,12 @@ def __init__(self, scheduler, **kwargs):
                 "comm_memory": [],
                 "comm_memory_limit": [],
                 "comm_buckets": [],
-                "comm_avg_duration": [],
-                "comm_avg_size": [],
```
Comment (crusaderky) [inline distributed/dashboard/components/scheduler.py:4330]: We're losing a little bit of functionality here.
IMHO it's not a big deal. Worth noting that we still have the information under the fine performance metrics (you'll have to calculate seconds/count and bytes/count yourself).

**Review by hendrikmakait — COMMENTED** [review]
I'd prefer to keep the existing bespoke metrics around for now. Some of them are more detailed and give us information per worker which has been extremely helpful in the past. Keeping them will also help us compare the information we get from the new approach and iterate if we identify gaps.

**Comment by crusaderky** [comment]
@hendrikmakait I've reinstated all metrics that are visible from the dashboard, as discussed. This is ready for review again.

**Review by hendrikmakait — APPROVED** [review]
Thanks, @crusaderky, this entire series of changes looks great!

**Pull request merged**